libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1693,12 +1700,12 @@ aggressiveInlining p
 
     go fvs sub p = case p of
       Subst x t p -> go fvs sub (substitutePred x t p)
-      Reify tm f b
-        | not (isLit tm)
-        , Lit a <- substituteAndSimplifyTerm sub tm -> do
+      Reifies t' t f
+        | not (isLit t)
+        , Lit a <- substituteAndSimplifyTerm sub t -> do
             tell $ Any True
-            pure $ mkReify (Lit a) f b
-        | otherwise -> Reify tm f <$> goBinder fvs sub b
+            pure $ Reifies t' (Lit a) f
+        | otherwise -> pure $ Reifies t' t f
```

Comment:
Do we have an argument, that eventually the second argument 't' will be a literal?

I assume the new version of simplifyPred, will try and make that happen?

Reply:
Yes, it will become a literal before solving because the dependency order specifies basically `dependsOn t t'`.